dev-libs/libprelude/libprelude-4.0.0.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6}} )
+
+USE_RUBY="ruby21 ruby22 ruby23 ruby24"
+
+inherit autotools python-r1 ruby-single
+
+DESCRIPTION="Prelude-SIEM Framework Library"
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="lua perl python ruby"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="net-libs/gnutls:=
+	lua? ( dev-lang/lua:* )
+	perl? ( dev-lang/perl:= virtual/perl-ExtUtils-MakeMaker )
+	python? ( ${PYTHON_DEPS} )
+	ruby? ( ${RUBY_DEPS} )"
+
+DEPEND="${RDEPEND}
+	sys-devel/flex
+	virtual/yacc
+	>=dev-lang/swig-3.0.7
+	virtual/pkgconfig"
+
+src_prepare() {
+	default
+
+	# Avoid null runpaths in Perl bindings.
+	sed -e 's/ LD_RUN_PATH=""//' -i "${S}/bindings/Makefile.am" || die "sed failed"
+
+	eautomake
+}
+
+src_configure() {
+	local python2_configure=--without-python2
+	local python3_configure=--without-python3
+
+	python_configure() {
+		if [[ ${EPYTHON} == python2* ]]; then
+			python2_configure=--with-python2
+		elif [[ ${EPYTHON} == python3* ]]; then
+			python3_configure=--with-python3
+		fi
+	}
+
+	if use python; then
+		python_foreach_impl python_configure
+	fi
+
+	econf \
+		--enable-easy-bindings \
+		--with-swig \
+		$(use_with lua) \
+		$(use_with perl) \
+		$(usex perl '--with-perl-installdirs=vendor' '' ) \
+		${python2_configure} \
+		${python3_configure} \
+		$(use_with ruby)
+}
+
+src_compile() {
+	default
+	python_compile() {
+		if [[ ${EPYTHON} == python2* ]]; then
```

Comment:
This kinda works, mostly because it uses distutils under the hood and you don't have any overlapping implementations on the list (e.g. pypy). However, what it does is semi-random and ugly in the end.

Since I did the research you were supposed to do in the first place, here's an idea. Do the `python_setup` dance described above, remove all `setup.py` calls from `bindings/python/Makefile.am` (but leave swig call there) and use distutils-r1 w/ `DISTUTILS_OPTIONAL=1` in the `bindings/python` subdirectory.

`econf` will take care of generating `setup.py`, `emake` will take care of running swig and distutils-r1 will do the actual build & install correctly, for every impl.

Reply:
> Since I did the research you were supposed to do in the first place

I did the research but with your previous help and the situation, it was really not clear. Thanks for digging a little and proposed and idea. I will look in this way.